File: Data Structures & Algorithms/two-integer-sum-ii/submission-0.rs
impl Solution {
    pub fn two_sum(numbers: Vec<i32>, target: i32) -> Vec<i32> {
        use std::cmp::Ordering;
        let mut i1: usize = 0;
        let mut i2: usize = numbers.len() - 1;

        while i2 > i1 {
            let sum = numbers[i1] + numbers[i2];
            match sum.cmp(&target) {
                Ordering::Less => { i1 += 1; },
                Ordering::Greater => { i2 -= 1; },
                Ordering::Equal => { return vec![i1 as i32 + 1, i2 as i32 + 1]; }
            }
        }
        vec![i1 as i32 + 1, i2 as i32 + 1]
    }
}

File: Data Structures & Algorithms/validate-parentheses/submission-0.py
class Solution:
    def isValid(self, s: str) -> bool:
        bc = [''] * 1000
        n = 0
        for b in s:
            if b== "[" or b== "(" or b== "{":
                bc[n] = b 
                n += 1
            else:
                if (b == ")" and bc[n-1] == "(") or (b == "]" and bc[n-1] == "[") or (b=="}" and bc[n-1] == "{"):
                    n -= 1
                else:
                    return False
        return n == 0